packages/client/src/comp/Address/Input.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react'
+import Input from 'muicss/lib/react/input'
+import styled from 'styled-components'
+import { BaseInput } from '../util/Input'
+import { Select, Option } from 'muicss/react'
+import { useControlRef } from '../util/ControlRef'
+import { allStates, AddressInputParts } from '../../common'
+import { useAppHistory } from '../../lib/path'
+import { cssQuery } from '../util/cssQuery'
+
+interface Props {
+  fields: AddressInputParts
+  setField: (id: keyof AddressInputParts, value: string) => void
+}
+
+const FlexBox = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  flex-flow: row wrap;
+  align-items: flex-end;
+  margin: 1em 0;
+`
+
+const Flex = styled.div<{ basis?: string, mobileBasis?: string }>`
+  flex-basis: ${p => p.mobileBasis ?? '100%'};
+  ${cssQuery.medium} { flex-basis: ${p => p.basis ?? '100%'}; }
+  /* Fixes needless scrollbars showing on Linux/Windows */
+  .mui-select label { overflow: hidden; }
+`
+
+// Since react-hooks/exhaustive-deps will trigger re-renders due to its
```

Comment:
Not really since `useLayoutEffect` is still going to require deps, the difference between using `useLayoutEffect` and `useEffect` in this case is that the former is going to fire before the DOM is painted while the later will fire after.